File: backend/models/audit.go
package models

import "time"

type AuditLog struct {
	ID        uint      `gorm:"primaryKey;column:id" json:"id"`
	Timestamp time.Time `gorm:"column:timestamp" json:"timestamp"`
	User      string    `gorm:"column:user" json:"user"`
	Action    string    `gorm:"column:action" json:"action"`
	Module    string    `gorm:"column:module" json:"module"`
	Details   string    `gorm:"column:details" json:"details"`
}

// TableName define o nome exato da tabela no banco
func (AuditLog) TableName() string {
	return "audit_logs"
}

File: backend-node/Models/Audit.js
module.exports = (sequelize, DataTypes) => {
  const AuditLog = sequelize.define('AuditLog', {
    id: { 
      type: DataTypes.INTEGER, 
      primaryKey: true, 
      autoIncrement: true 
    },
    table_name: { 
      type: DataTypes.STRING 
    },
    action: { 
      type: DataTypes.STRING, 
      allowNull: false 
    },
    record_id: { 
      type: DataTypes.INTEGER 
    },
    old_data: { 
      type: DataTypes.TEXT // TEXT é mais seguro aqui para evitar problemas de codificação do JSON
    },
    new_data: { 
      type: DataTypes.TEXT 
    },
    changed_at: { 
      type: DataTypes.DATE 
    },
    timestamp: { 
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW
    },
    user: { 
      type: DataTypes.STRING 
    },
    module: { 
      type: DataTypes.STRING 
    },
    details: { 
      type: DataTypes.TEXT 
    }
  }, { 
    tableName: 'audit_logs', 
    timestamps: false // Deixamos false porque já mapeamos 'changed_at' e 'timestamp' manualmente acima
  });

  return AuditLog;
};